core/src/main/java/org/elasticsearch/index/translog/TranslogDeletionPolicy.java

```diff
@@ -76,6 +81,17 @@ public synchronized void setMinTranslogGenerationForRecovery(long newGen) {
         minTranslogGenerationForRecovery = newGen;
     }
 
+    /**
+     * Sets the translog generation of the last index commit.
+     */
+    public synchronized void setTranslogGenerationOfLastCommit(long lastGen) {
+        if (lastGen < translogGenerationOfLastCommit) {
```

Comment:
it would be nice to also check it's higher than the "required for recovery gen".